src/main/java/uk/gov/hmcts/darts/arm/rpo/impl/ArmRpoApiImpl.java

```diff
@@ -59,8 +63,37 @@ public void getRecordManagementMatter(String bearerToken, Integer executionId, U
     }
 
     @Override
-    public IndexesByMatterIdResponse getIndexesByMatterId(String bearerToken, Integer executionId, UserAccountEntity userAccount) {
-        throw new NotImplementedException("Method not implemented");
+    public void getIndexesByMatterId(String bearerToken, Integer executionId, String matterId, UserAccountEntity userAccount) {
+        var armRpoExecutionDetailEntity = armRpoService.getArmRpoExecutionDetailEntity(executionId);
+        armRpoService.updateArmRpoStateAndStatus(armRpoExecutionDetailEntity, ArmRpoHelper.getIndexesByMatterIdRpoState(),
+                                                 ArmRpoHelper.inProgressRpoStatus(), userAccount);
+
+        StringBuilder errorMessage = new StringBuilder("Failure during ARM RPO get indexes by matter ID: ");
+        IndexesByMatterIdResponse indexesByMatterIdResponse;
+        try {
+            indexesByMatterIdResponse = armRpoClient.getIndexesByMatterId(bearerToken, createIndexesByMatterIdRequest(matterId));
+        } catch (FeignException e) {
+            // this ensures the full error body containing the ARM error detail is logged rather than a truncated version
+            log.error(errorMessage.append("Unable to get ARM RPO response") + " {}", e.contentUTF8());
+            throw handleFailureAndCreateException(errorMessage.toString(), armRpoExecutionDetailEntity, userAccount);
+        }
+
+        if (isNull(indexesByMatterIdResponse)
+            || CollectionUtils.isEmpty(indexesByMatterIdResponse.getIndexes())
+            || isNull(indexesByMatterIdResponse.getIndexes().getFirst())) {
+            throw handleFailureAndCreateException(errorMessage.append("Unable to find indexes by matter ID in response").toString(),
+                                                  armRpoExecutionDetailEntity,
+                                                  userAccount);
+        }
+
+        armRpoExecutionDetailEntity.setIndexId(indexesByMatterIdResponse.getIndexes().getFirst().getIndex().getIndexId());
```

Comment:
Worth checking if `index` is null, to guard against NPE when invoking `getIndexId()`?

Reply:
done in line 82

Reply:
I mean if we get a response like this:

```json
{
  "indexes": [
    {
      "index": null,
      "isMultiStream": false,
      "children": []
    }
  ]
}
```

The check on L82 will pass as `indexes` is not null or empty, but if the nested `index` is null (or omited from the response) then you'll get NPE when you try to get the `indexId` attribute of the null `index` object.